test-utils/src/random.rs

```diff
@@ -39,6 +39,12 @@ impl Seed {
         self.0
     }
 
+    /// In tests where multiple seeds are needed, this can be used to deterministically create a new, different seed
+    pub fn derive_seed(&self) -> Self {
+        let new_seed = ChaChaRng::seed_from_u64(self.as_u64()).next_u64();
+        Self(new_seed)
+    }
```

Comment:
This may be potentially confusing for it generates the same seed every time and so has to be used in a linear way. I'd prefer slapping the `Distribution<Seed>` on `rand::distributions::Standard`, so you can generate new seeds using existing RNGs by calling `rng.gen::<Seed>()`.

Reply:
Done